Maven HintsPanel UI layout fixes and minor cleanup.

mbien opened the pull request:
- analog to #3472, this fixes UI component resize behavior of the Maven HintsPanel
 - description panel is now filling the space vertically like in the java version of this panel. Text is now wrapping properly.
 - the option panels had to be adjusted to prefer their minimum default size
 - minor language cleanup: diamonds, lambdas

targets delivery. But this is fairly low priority. Noticed those UI issues while writing the maven release version hint.


![hints-before](https://user-images.githubusercontent.com/114367/199617875-d3572946-1bd3-4f29-b73d-ce9e62f2c155.png)

![hints-after](https://user-images.githubusercontent.com/114367/199617898-9de3461a-e359-4bf1-bc6a-2cbb9e96a196.png)

sdedic commented:
Just to be sure I've not overlooked anything: no functional change, just layout & cleanup anonymous -> lambda ?

mbien commented:
> Just to be sure I've not overlooked anything: no functional change, just layout & cleanup anonymous -> lambda ?

right. this is primarily about UI layouts. The lamda cleanup is just a cleanup, no logic changed there.

mbien commented:
added another commit for the CSL HintsPanel. All three panels are now updated. They also have the same initial divider locations and look like the same component.

neilcsmith-net commented:
Weird thing is the issue isn't reproducible with a clean userdir until the options window is resized.

Makes sense, and as we're having another rc, let's merge.
